### DISCUSSION TITLE
Longwave radiation budget problem in NoahMP LSM

### DISCUSSION THREAD

**Discussion by sudarsan3010** [NCAR/hrldas#195] [Q&A] [answered]
**I am using ERA-5 Land forcing with the following model configurations,**

 DYNAMIC_VEG_OPTION                = 5
 CANOPY_STOMATAL_RESISTANCE_OPTION = 1
 BTR_OPTION                        = 1
 SURFACE_RUNOFF_OPTION             = 1
 SUBSURFACE_RUNOFF_OPTION          = 1
 DVIC_INFILTRATION_OPTION          = 1
 SURFACE_DRAG_OPTION               = 1
 FROZEN_SOIL_OPTION                = 1
 SUPERCOOLED_WATER_OPTION          = 1
 RADIATIVE_TRANSFER_OPTION         = 1
 SNOW_ALBEDO_OPTION                = 2
 PCP_PARTITION_OPTION              = 1
 SNOW_THERMAL_CONDUCTIVITY         = 1
 TBOT_OPTION                       = 2
 TEMP_TIME_SCHEME_OPTION           = 1
 GLACIER_OPTION                    = 1
 SURFACE_RESISTANCE_OPTION         = 1
 SOIL_DATA_OPTION                  = 1
 PEDOTRANSFER_OPTION               = 1
 CROP_OPTION                       = 0
 IRRIGATION_OPTION                 = 0
 IRRIGATION_METHOD                 = 0
 TILE_DRAINAGE_OPTION              = 0

 FORCING_TIMESTEP = 3600
 NOAH_TIMESTEP    = 900
 OUTPUT_TIMESTEP  = 86400

 SPLIT_OUTPUT_COUNT = 1
 SKIP_FIRST_OUTPUT = .false.
 RESTART_FREQUENCY_HOURS = -1

**I have been trying to run the model for one year, but the model stops after a month (up to 20080123100) and gets stopped by giving the following error.**

***DATE=2008-01-31_23:00:00 284.85693   0.95198    Timing:   0.07 Cumulative:      274.84  SFLX:   0.06
 ***DATE=2008-01-31_23:00:00 280.09055   2.24266    Timing:   0.07 Cumulative:      274.84  SFLX:   0.06
 ***DATE=2008-01-31_23:15:00    Timing:   0.08 Cumulative:      274.88  SFLX:   0.03
 emitted longwave <0; skin T may be wrong due to inconsistent
 input of VegFracGreen with LeafAreaIndex
 VegFrac =    0.00000000     VegAreaIndEff =    0.00000000     TemperatureCanopy =    239.137390     TemperatureGrd =    111.317917
 RadLwDownRefHeight =   -176.621796     RadLwNetSfc =    176.037598     SnowDepth =    4.34467363
Note: The following floating-point exceptions are signalling: IEEE_UNDERFLOW_FLAG IEEE_DENORMAL
STOP Error: Longwave radiation budget problem in NoahMP LSM
 ***DATE=2008-01-31_23:15:00 285.39563   2.38023    Timing:   0.09 Cumulative:      274.90  SFLX:   0.04

**Comment by RAbolafiaRosenzweig** [comment]
Can you please check to make sure your input surface air temperature forcing is in units of Kelvin and varies within a physically realistic range? It looks like the canopy temperature is extremely cold (~ -34C) and the ground temperature is unrealistic (~ -162C), which makes me think that the surface air temp. forcing may not be in Kelvin. When you do this check, please also consider the other meteorological forcing variables and the corresponding units.

**Reply by sudarsan3010** [reply]
I have cross-checked the T2D and other meteorological forcings; every variable is in its range and units are appropriate. I have not changed anything from the Vtable. My study domain also covers the Himalayan region, with snow and glaciers.

![image](https://github.com/NCAR/hrldas/assets/53935951/5d30ca44-e9fa-4b7b-b489-f89ee80ff6d6)

From the last output obtained (at 2008013100), I am sharing below the screenshot of Ground Temperature (TG) and Canopy Air Temperature (TAH)

**Ground Temperature (TG)**
![image](https://github.com/NCAR/hrldas/assets/53935951/6833a791-82c0-47ee-b442-7d4fc99c85c2)

**Canopy Air Temperature (TAH)**
![image](https://github.com/NCAR/hrldas/assets/53935951/c078ea60-6fec-4089-8602-80ca3791e313)

Is this error because I am using the hrldas-master from github, is there some changes in code in this version? 

                                            --------------xxxx------------------xxxx-------------

2.  [Hrldas-v5.0.0](https://github.com/NCAR/hrldas/releases/tag/v5.0.0), I am getting the following error. The model not even gives out the initial time step output, stops before that.

dst(1) = src(1)* 0.89286 + src(2)* 0.10714
layer_bottom(1:nsoil) =    0.0700   0.2800   1.0000   2.5500
layer_top(1:nsoil)    =    0.0000   0.0700   0.2800   1.0000
Soil depth =     0.100000    0.300000    0.600000    1.000000
 k, dst_centerpoint(k) =            1   5.00000007E-02
 k, dst_centerpoint(k) =            2  0.250000000
 k, dst_centerpoint(k) =            3  0.700000048
 k, dst_centerpoint(k) =            4   1.50000000

 k, src_centerpoint(k) =            1   3.50000001E-02
 k, src_centerpoint(k) =            2  0.174999997
 k, src_centerpoint(k) =            3  0.639999986
 k, src_centerpoint(k) =            4   1.77499998
dst(1) = src(1)* 0.89286 + src(2)* 0.10714
At line 1725 of file module_hrldas_netcdf_io.f90
Fortran runtime error: Array bound mismatch for dimension 2 of array 'var' (4/32)

**Comment by tslin2** [comment]
maybe try using RADIATIVE_TRANSFER_OPTION = 2 or RADIATIVE_TRANSFER_OPTION = 3

**Reply by sudarsan3010** [reply]
I have tried using all the 3 options of Radiative transfer. It's still the same issue.

I have also changed the soil thickness to match with the Era-5 Land initialization, even that doesn't solve the issue.

**Comment by cenlinhe** [comment]
longwave error often occurs due to weird forcing values. My guess is that you have some unresolved coast points without any forcing values which caused the model crash.
If you used HRLDAS_forcing to generate your initial and forcing data, please try to increase the value of EXPAND_LOOP (e.g., to 10) in HRLDAS_forcing when you generate the initial and forcing data and re-run the model again.

**Reply by sudarsan3010** [reply]
I already reviewed the discussion threads and in the HRLDAS_forcing configuration, I used EXPAND_LOOP=10. I will check by increasing it further and update it here.
![image](https://github.com/NCAR/hrldas/assets/53935951/067143be-eda2-4d88-9da3-1e2c747e142e)

**Comment by ZhaoY-github** [comment]
It's likely that you are using a land use and land cover dataset that contains lake category, for example, MODIS lake=21, while your IST is still 1 (identified as soil) at certain grid cells, which makes the model confused. 
Make sure XLAND in wrfinput and LANDMASK and LU_INDEX in geo_em.d01.nc are consistent. You could manual adjust the values in XLAND.
@cenlinhe any suggestion?

**Comment by sudarsan3010** [comment, marked as answer]
I realized my error. When using 'Prepare_ERA5-Land_Forcing.sh', it expects one file per year, but I processed it for each month, so the model terminated exactly at the end of the month. 

Anyway, I have increased it to EXPAND_LOOPS=20 to overcome further errors. 
Thank you all for your timely responses and suggestions. :)

**Reply by sreehari2488** [reply]
I have experienced the same error but i only used a month of era5 forcing data. How did you fix the error?

**Reply by sudarsan3010** [reply]
> I have experienced the same error but i only used a month of era5 forcing data. How did you fix the error?

Once the multiple month's data are downloaded merge the monthly grid to yearly grib
cdo -O cat ERA5_global_2008*.grib ERA5_global_2008.grib

Yearly forcing grib files are created in this way. And then use this yearly grib in 'Prepare_ERA5-Land_Forcing.sh'

**Comment by neha12-3** [comment]
i am also facing little bit similar issue.can anyone plz help 
![Uploading IMG_20241025_130708.jpg…]()

**Reply by sudarsan3010** [reply]
I think your image is not uploaded properly. It is not visible.

**Comment by neha12-3** [comment]
I have deaccumulate rainfall data
